digitalocean/datasource_digitalocean_regions.go

```diff
@@ -0,0 +1,60 @@
+package digitalocean
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/digitalocean/godo"
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func dataSourceDigitalOceanRegions() *schema.Resource {
+	return &schema.Resource{
+		Read:   dataSourceDigitalOceanRegionsRead,
+		Schema: map[string]*schema.Schema{},
```

Comment:
My initial thought would be for the schema to look something like:
```suggestion
		Schema: map[string]*schema.Schema{
			"regions": {
				Type:     schema.TypeSet,
				Computed: true,
				Elem: &schema.Resource{
					Schema: map[string]*schema.Schema{
						"name": {
							Type:     schema.TypeString,
							Computed: true,
						},
						"slug": {
							Type:     schema.TypeString,
							Computed: true,
						},
					},
				},
			},
		},
```

Which makes the UI like:

```
data "digitalocean_regions" "foobar" {
}

resource "digitalocean_droplet" "web" {
  image  = "ubuntu-18-04-x64"
  name   = "web-1"
  region = "nyc2"
  size   = digitalocean_regions.foobar.regions[0].slug
}

output "digitalocean_regions" {
   value = digitalocean_regions.foobar.regions
}

# digitalocean_regions = [
#   {
#     "name" = "New York 1"
#     "slug" = "nyc1"
#   },
#   {
#     "name" = "New York 2"
#     "slug" = "nyc2"
#   },
#  # and so on...
# ]
```

I think this matches the API more closely and makes it more extensible for any additional information about the regions that might be useful.